### PR TITLE
Add extra DefaultServlet include tests

### LINKED ISSUE #10142: req.getRequestDispatcher("/test.htm").include(req, resp); does not go anywhere on jetty12ee10 on exploded deploy, works on ee9 and older jettys.
**Jetty version(s)**
jetty12 built from source
**Java version/vendor** `(use: java -version)`
JDK 17
**OS type/version**
Windows/Unix
**Description**
Maybe the same as https://github.com/eclipse/jetty.project/issues/10139 ?
**How to reproduce?**
I have a simple servlet that does forwarding into a local htm file:

``` java
	protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException {
		System.out.println("MyServlet.doGet() called");
		try {
		  req.getRequestDispatcher("/test.htm").include(req, resp);
		} catch (IOException e) {
			throw new ServletException(e);
		}
	}
```

test.htm is in the root of the web app.
If I deploy the application as packaged application on jetty12ee10 then it works and test.htm is shown. But If I deploy it as exploded application then I will get an empty page. The same application for jett12ee9 it works in both modes.

### PR DESCRIPTION
**Pull request by lorban**
It's actually #10140 that fixes #10142, this PR merely adds some extra tests.

### CODE REVIEW

**Comment by joakime** [comment]
@lorban can you fix the title of this PR to have it make more sense when it shows up in changelog / release notes / VERSION.txt

File: jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/DefaultServlet.java [hunk 1]
```diff
@@ -1225,7 +1225,7 @@ private static boolean isIncluded(HttpServletRequest request)
 
     private static boolean isDefaultMapping(HttpServletRequest req)
     {
-        return req.getHttpServletMapping().getMappingMatch() == MappingMatch.DEFAULT;
+        return isIncluded(req) || req.getHttpServletMapping().getMappingMatch() == MappingMatch.DEFAULT;
```
Comment (gregw) [inline jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/DefaultServlet.java]: I don't think this is sufficient/correct.     A `DefaultServlet` mapped to some other location and/or name can be the target of an include.

I'm beginning to wonder if we should rename this to be `ResourceServlet` with this method returning false, and then just have an extended version of it called `DefaultServlet` that returns true for this method?

**Review by gregw — CHANGES_REQUESTED** [review]

File: jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/DefaultServlet.java (hunk 1 above)
Comment (gregw) [reply jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/DefaultServlet.java]: hmmm the problem with the idea above is that the same servlet instance might be both at "/" and "/somepath/*".
Oh, what happens if it is mapped to "*.suffix"? Do we use the right part of the path then?

Comment (gregw) [reply jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/DefaultServlet.java]: Maybe we just use the `pathInfo` if the `servletPath` is empty?

File: jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/DefaultServlet.java [hunk 2]
```diff
@@ -1228,7 +1228,7 @@ private static boolean isDefaultMappingIncluded(HttpServletRequest req)
         return req.getAttribute(RequestDispatcher.INCLUDE_PATH_INFO) == null;
     }
 
-    private static boolean isDefaultMapping(HttpServletRequest req)
+    private boolean isDefaultMapping(HttpServletRequest req)
```
Comment (gregw) [inline jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/DefaultServlet.java]: Let's make this protected so users can change this "heuristic"

**Review by gregw — CHANGES_REQUESTED** [review]

**Review by gregw — APPROVED** [review]

**Review by joakime — APPROVED** [review]

**Pull request merged**